### DISCUSSION TITLE
Gap in frequency domain

### DISCUSSION THREAD

**Discussion by Navids94** [geodynamics/sw4#227] [General]
Hi,
Synthetic and recorded velocity spectrum amplitudes differ by a factor of 100 at soil sites. I used point source for an small  (Mw~3.4) earthquake(source x=3200 y=38157 z=14500 strike=313 rake=32 dip=30 m0=3.05492e+14 type=VerySmoothBump freq=6 t0=3
prefilter fc2=6.0 type=lowpass passes=2 order=2)
I would appreciate it if someone could help me identify the reason for this frequency gap.

**Comment by andersp** [comment]
Is your grid fine enough to resolve frequencies up to 6 Hz? Did you follow the instructions in Section 4.4 of the user's guide?

**Reply by Navids94** [reply]
Thank you for your reply. Yes, I followed the instructions. Since I used VerySmoothBump, the value of f0 is equal to the regular corner frequency of the earthquake, and based on my calculations, the corner frequency for this earthquake is around 6. Therefore, I set freq to 6. Additionally, I obtained the following Vsmin/h from the SW4 reports (my trigger frequency is 5 Hz):

***** PPW = minVs/h/maxFrequency ********
g=0, h=6.400000e+01, minVs/h=54.375 (Cartesian)
g=1, h=3.200000e+01, minVs/h=111.875 (Cartesian)
g=2, h=1.600000e+01, minVs/h=42.0725 (Cartesian)
g=3, h=8.000000e+00, minVs/h=40.755 (Cartesian)
g=4, h=4.000000e+00, minVs/h=45.25 (Cartesian)

*** Attenuation parameters calculated for 3 mechanisms,
      max freq=1.000000e+01 [Hz], min_freq=1.000000e-01 [Hz], velo_freq=1.000000e+00 [Hz]

So, the minimum minVs/h is 40.755. If we divide this by 8 ppw, we have 5 Hz.

**Reply by andersp** [reply]
Why are you using 5 instead of 6 Hz when evaluating the resolution? The resolution requirements should be applied to the source frequency. Also, I would switch to a Gaussian time function, which has better smoothness properties. How are you saving the velocity traces from your simulation? With this setup the primary state variable correspond to particle velocities, so no additional differentiation should be applied when saving the results.

**Comment by Navids94** [comment]
I need a Frequency range of 0.1 Hz to 5 Hz, so based on output I have minVs/h=40.755 and by dividing this by fmax I need(5Hz) I have ppw=8. Is this correct? I set f0 = freq = 6 Hz in the source command because the frequency of this earthquake is 6 Hz(I found fc=6 on paper, which relates to earthquake frequency, not calculated based on the SW4 manual). 
I also chose fc2 = 6 Hz in the prefilter to consider a value slightly larger than the 5 Hz I need. Please correct me if I am wrong.

**Comment by Navids94** [comment]
@andersp I used **C6SmoothBump** and **VerySmoothBump**, which generate **velocity** since both start and end at zero. Thank you for your suggestion, I will try to use Gaussian time function.
I think as long as the minimum minVs/h is over **40**, I will get **5Hz** resolution regardless of the corner frequency in the source command, as sometimes I change the f0 in source to observe the effects of corner frequency on simulation.

**Reply by andersp** [reply]
@Navids94 In the user's guide there is some discussion about the frequency content triggered by the VerySmoothBump source time function. I believe it can trigger frequencies up to 4*f0. You would be better off using the Gaussian, but it is important to set t0 >= 6/omega, where omega = 2*pi*f0. Then the tail of the Gaussian can be neglected. Regarding the resolution, you need to pick the grid size to match the frequency band triggered by the source, or the corner frequency in the prefilter command. Otherwise you will get spurious numerical modes on the grid. This is unrelated to what frequency band you want to study.

**Reply by Navids94** [reply]
@andersp  Based on my calculations using Brune's model for my simulation earthquake with a moment magnitude of **3.4**, the corner frequency ranges between **1 and 2.5 Hz**. If I set **f0= 2 Hz(fmax ~ 2.5f0 ~2.5 * 2 ~ 5Hz) in source command** and **fc2=5.** in the prefilter command, while refining the mesh grid size to ensure that the minimum **Vs/h ratio is greater than or equal to 40,** I think this will produce a good match.
So, for the next run, I will be using the following command:
source x=3200 y=38157 z=14500 strike=313 rake=32 dip=30 **m0=3.05492e+14 type=Gaussian freq=12.57 (12.57/2pi=2Hz) t0=2**
prefilter **fc2=5. type=lowpass passes=2 order=2**

I think it's fine. Thank you.
